services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/UpdateServerSettingsFragment.java

```diff
@@ -0,0 +1,152 @@
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.google.android.material.textfield.TextInputLayout;
+
+import org.opendatakit.consts.IntentConsts;
+import org.opendatakit.properties.CommonToolProperties;
+import org.opendatakit.properties.PropertiesSingleton;
+import org.opendatakit.services.R;
+import org.opendatakit.services.sync.actions.activities.VerifyServerSettingsActivity;
+import org.opendatakit.services.sync.actions.viewModels.AbsSyncViewModel;
+import org.opendatakit.services.utilities.UserState;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UpdateServerSettingsFragment extends Fragment {
+
+    private class OnButtonClick implements View.OnClickListener {
+        @Override
+        public void onClick(View v) {
+            if (v.getId() == R.id.btnUpdateServerUrl) {
+                onClickUpdateUrl();
+            } else if (v.getId() == R.id.btnChooseDefaultServer) {
+                inputServerUrl.getEditText().setText(getString(R.string.default_sync_server_url));
+            } else if (v.getId() == R.id.btnVerifyServerUpdateServerDetails) {
+                startVerifyActivity();
+            } else if (v.getId() == R.id.btnScanQrUpdateServerDetails) {
+                // TODO
+            }
+        }
+    }
+
+    private TextInputLayout inputServerUrl;
+    private AbsSyncViewModel absSyncViewModel;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_update_server_settings, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        findViewsAndAttachListeners(view);
+        setupViewModelAndNavController();
+    }
+
+    private void findViewsAndAttachListeners(View view) {
+        inputServerUrl = view.findViewById(R.id.inputServerUrl);
+        Button btnUpdateUrl = view.findViewById(R.id.btnUpdateServerUrl);
+        Button btnSetDefault = view.findViewById(R.id.btnChooseDefaultServer);
+        Button btnVerifyServerDetails = view.findViewById(R.id.btnVerifyServerUpdateServerDetails);
+        Button btnScanQr = view.findViewById(R.id.btnScanQrUpdateServerDetails);
+
+        OnButtonClick onButtonClick = new OnButtonClick();
+        btnUpdateUrl.setOnClickListener(onButtonClick);
+        btnSetDefault.setOnClickListener(onButtonClick);
+        btnVerifyServerDetails.setOnClickListener(onButtonClick);
+        btnScanQr.setOnClickListener(onButtonClick);
```

Comment:
I've commented about this in my another PR review make sure you fix it in here also.